### PR TITLE
Improve text of README.md

### PR DESCRIPTION
**Pull request by b-scholz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -366,16 +351,15 @@ func ExampleDatabase_QueryHistoricState() {
 
 # Performance 
 
-Carmen is the backbone of [Fantom Sonic](https://fantom.foundation/sonicPage) blockchain. 
-It is the key component that enables high performance in a disk space conservative manner. 
+Carmen is the StateDB of [Fantom Sonic](https://fantom.foundation/sonicPage) blockchain. 
+The data store is the key component for enabling low time-to-finally and high transaction throughput while saving disk space.
```
Comment (petr-hanzl) [inline README.md:355]:
```suggestion
The data store is the key component for enabling low time-to-finality and high transaction throughput while saving disk space. 
```

**Review by petr-hanzl — APPROVED** [review]
Thanks for improving the readme!

File: README.md [hunk 2]
```diff
@@ -277,8 +263,7 @@ func ExampleDatabase_GetHistoricContext() {
 
 
 ## Query API
-Both LiveDB and Archive can be queried using a **query API** that avoids hassle packing every request to a block 
-and transaction in previous APIs. It is done via a functional style callback that provides a query:
+LiveDB and Archive can be queried using a **query API** using a functional style callback as shown below:
```
Comment (kjezek) [inline README.md:266]:
```suggestion
LiveDB and Archive can be queried using a **query API** providing a functional style callback as shown below:
```

Comment (kjezek) [reply README.md:266]: avoided `using` twice in one sentence

File: README.md [hunk 3]
```diff
@@ -36,21 +26,17 @@ import "github.com/Fantom-foundation/Carmen/go/carmen"
 
 
 # How to Use
-Carmen is configured for the scheme to use, and if historical data are retained or not. The Merkle-Patricia-Trie 
-compatible scheme is called S5, and it can be either Archive or non-Archive (LiveDB).
+Schema S5 is the only schema compatible with Merkle-Patricia Tries. Carmen has always enabled LiveDB to keep the state of the last block.
```
Comment (kjezek) [inline README.md:29]:
```suggestion
Schema S5 is the only schema compatible with Merkle-Patricia Tries. Carmen enables always LiveDB to keep the state of the last block.
```

File: README.md [hunk 4]
```diff
@@ -36,21 +26,17 @@ import "github.com/Fantom-foundation/Carmen/go/carmen"
...
 
-A state of the blockchain may be updated always only at the head of the chain, and it is stored in the LiveDB. 
-The state becomes verbatim as part of a block once appended to the blockchain via the new block. If the Archive mode 
-is enabled, the state of the last block is appended to the archive and it can be retrieved later. If the Archive 
-is not configured, only the head state is available. LiveDB is enabled all the time, while the Archive is tentative.
-
-As a blockchain is a sequence of blocks where each contains a set of transactions, the API is oriented 
-to creation and retrieval of transactions within blocks. Carmen provides an API for accessing both Live 
-and Archive databases to append new blocks to the blockchain and to query history for previous blocks.
+The Carmen interface is loosely related to the go-ethereum's StateDB interface, though it differs w.r.t. block management and transaction management.
```
Comment (kjezek) [inline README.md:32]: not sure if this change improves this paragraph. First it IMHO unnecessarily refers to ethereum's StateDB and then suddenly it becomes too brief

**Review by petr-hanzl — APPROVED** [review]

File: README.md [hunk 5]
```diff
@@ -1,16 +1,6 @@
...
-quality schema is state hash root compatible with Ethereum’s World State, organised in the Merkle Patricia Trie.
-
-A few experimental schemas are provided that utilise fast flat storages with various organisation
-of key value pairs. A version implemented in Golang and C++ is provided for a selection of schemas.
+Carmen is a high-performance data store for Blockchains and manages balances, code, and the persistent storage of smart contracts. Carmen assumes a linear evolution of blocks. Carmen supports two versions of databases: LiveDB, which keeps the last state of the last block only, and ArchiveDB, which keeps all states over all blocks. The Carmen project implements various schemas (currently S1 to S5), including a Merkle-Patricia Tries (MPT) variation as an underlying data structure for storing state.  The storage layer is abstracted, and the schemas can read and write information in memory, a key-value store (LevelDB, etc.), or a native file format. Carmen has implemented schema S1-S3 in the Go and C++ languages. Schema S4 and S5 are implemented only in Go.  The latest Schema S5 is currently used in the Sonic Client.
```
Comment (kjezek) [inline README.md:3]:
```suggestion
Carmen is a high-performance data store for Blockchains and manages balances, code, and the persistent storage of smart contracts. Carmen assumes a linear evolution of blocks. Carmen supports two versions of databases: LiveDB, which keeps the last state of the last block only, and ArchiveDB, which keeps all states over all blocks. The Carmen project implements various schemas (currently S1 to S5), including a Merkle-Patricia Trie (MPT) variation as an underlying data structure for storing state.  The storage layer is abstracted, and the schemas can read and write information in memory, a key-value store (LevelDB, etc.), or a native file format. Carmen has implemented schema S1-S3 in the Go and C++ languages. Schema S4 and S5 are implemented only in Go.  The latest Schema S5 is currently used in the Sonic Client.
```

Comment (kjezek) [reply README.md:3]: Could you also add line separators so it is not a one long line?

File: README.md [hunk 6]
```diff
@@ -1,16 +1,6 @@
 # Introduction
 
-Carmen is a fast and space conservative database for blockchains. It outperforms other projects in the
```
Comment (kjezek) [inline README.md:3]: I think this two original lines should be (in some form) kept. It is our pitch - the speed that outperforms others and the space is a fraction of what provides others. 

The new pitch is way too complicated, it suddenly speaks about balances, code, smart contract.

Comment (b-scholz) [reply README.md:3]: @kjezek  - can you rewrite it and submit it? I can check after.

Comment (kjezek) [reply README.md:3]: done, please have a look

Comment (b-scholz) [reply README.md:3]: We should focus on functionality rather than non-functional properties.

**Review by kjezek — APPROVED** [review]

**Review by petr-hanzl — APPROVED** [review]

**Pull request merged**